### ISSUE TITLE
Healthcheck endpoint?

### ISSUE THREAD

**Issue by JohnGalt1717** [ggerganov/llama.cpp#4746]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [ X] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [ X] I carefully followed the [README.md](https://github.com/ggerganov/llama.cpp/blob/master/README.md).
- [ X] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [ X] I reviewed the [Discussions](https://github.com/ggerganov/llama.cpp/discussions), and have a new bug or useful enhancement to share.

# Feature Description
I'd like the server to have a /health endpoint.  

# Motivation

Basically this is for docker so that I can restart the server if it fails automatically and recover given that llama.cpp will crash the process in many cases.

# Possible Implementation
/health as a get that will return 200 if it's healthy and obviously timeout if the server isn't available.  The best I can do right now is props

Also, the final docker container needs to have curl or wget installed into it, and then documentation updated to show how to go and use the docker-compose functionality to do this.

I'd also like to see /completion and the async version return a 429 error instead of 404 when it is busy as 429 is easily retried but 404 is a not found and thus terminal.

**Comment by Huge** [comment]
https://github.com/ggerganov/llama.cpp/pull/4853 seems related, no clue why that PR is "closed".

**Comment by Celarye** [comment]
These have been added through #4881 instead.

**Comment by bsquizz** [comment]
There's an issue with the llama-server.Dockerfile -- curl is not installed in the final runtime layer, so the health check can not be run from inside the image. PR open here to fix it: https://github.com/ggerganov/llama.cpp/pull/8693
